### PR TITLE
fix: random zones only when zones are not provided

### PR DESCRIPTION
**Pull request by ericyz**
Solves https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/1442

### CODE REVIEW

File: examples/simple_zonal_with_acm/acm.tf [hunk 1]
```diff
@@ -28,7 +28,7 @@ module "acm" {
 
   secret_type = "ssh"
 
-  policy_bundles = ["https://github.com/GoogleCloudPlatform/acm-policy-controller-library/bundles/policy-essentials-v2022?ref=59f4695394285078f7c2029ec7d0f9ed1d6d700a"]
+  policy_bundles = ["https://github.com/GoogleCloudPlatform/acm-policy-controller-library/bundles/policy-essentials-v2022?ref=dd3f932eefa4c4c44c548144be1c2331d8594689"]
```
Comment (bharathkkb) [inline examples/simple_zonal_with_acm/acm.tf:31]: Was there a reason for this change?

**Review by bharathkkb — APPROVED** [review]
Thanks for the PR @ericyz. A minor comment about the ACM change.

File: examples/simple_zonal_with_acm/acm.tf (hunk 1 above)
Comment (ericyz) [reply examples/simple_zonal_with_acm/acm.tf:31]: The pipeline was failing - https://console.cloud.google.com/cloud-build/builds;region=global/b022a62b-5437-4951-9733-e491b2a82248?project=cloud-foundation-cicd

unknown field "namespace" in 
sh.gatekeeper.constraints.v1beta1.K8sRestrictRoleBindings.spec.parameters.allowedSubjects];

**Pull request merged**